ic-os/components/networking/setup-networking/setup-networking.sh

```diff
@@ -0,0 +1,138 @@
+            "ipv6_gateway") IPV6_GATEWAY="${value}" ;;
+        esac
+    done <"${CONFIG}"
+
+    if [ -z "$IPV6_GATEWAY" ]; then
+        echo "No IPv6 gateway found in $CONFIG."
+        exit 1
+    fi
+}
+
+function generate_addresses() {
+    echo "Generating MAC and IPv6 addresses..."
+    IC_BIN_PATH="${IC_BIN_PATH:-/opt/ic/bin}"
+
+    if [ "$NODE_TYPE" = "SetupOS" ]; then
+        MAC_ADDR=$("${IC_BIN_PATH}/setupos_tool" generate-mac-address --node-type SetupOS)
+        IPV6_ADDR=$("${IC_BIN_PATH}/setupos_tool" generate-ipv6-address --node-type SetupOS)
+    else
+        MAC_ADDR=$("${IC_BIN_PATH}/hostos_tool" generate-mac-address --node-type HostOS)
+        IPV6_ADDR=$("${IC_BIN_PATH}/hostos_tool" generate-ipv6-address --node-type HostOS)
+    fi
+
+    if [ -z "$MAC_ADDR" ] || [ -z "$IPV6_ADDR" ]; then
+        echo "Failed to generate MAC or IPv6 address."
+        exit 1
+    fi
+
+    echo "Generated MAC address: $MAC_ADDR"
+    echo "Generated IPv6 address: $IPV6_ADDR"
+}
+
+function gather_interfaces_by_speed() {
+    echo "Gathering and sorting interfaces by speed..."
+
+    SYS_NET_PATH="${SYS_NET_PATH:-/sys/class/net}"
+    INTERFACES=($(find "$SYS_NET_PATH" -type l -not -lname '*virtual*' -exec basename '{}' ';'))
+    declare -A SPEED_MAP
+
+    for IFACE in "${INTERFACES[@]}"; do
+        if [ -f "$SYS_NET_PATH/$IFACE/speed" ]; then
+            SPEED=$(cat "$SYS_NET_PATH/$IFACE/speed" 2>/dev/null || echo 0)
+        else
+            SPEED=0
+            echo "Warning: Unable to determine speed for interface $IFACE, defaulting to 0." >&2
+        fi
+        SPEED_MAP["$IFACE"]=$SPEED
+    done
+
+    # Sort interfaces by speed descending
+    SORTED_INTERFACES=$(for IFACE in "${!SPEED_MAP[@]}"; do
+        echo "${SPEED_MAP[$IFACE]} $IFACE"
+    done | sort -nrk1 | awk '{print $2}')
+
+    if [ -z "$SORTED_INTERFACES" ]; then
+        echo "No interfaces found."
+        exit 1
+    fi
+
+    INTERFACE_LIST=$(echo "$SORTED_INTERFACES" | paste -sd, -)
+    echo "Interfaces sorted by speed: $INTERFACE_LIST"
+}
+
+function configure_netplan() {
+    echo "Configuring netplan..."
+    NETPLAN_TEMPLATE_PATH="${NETPLAN_TEMPLATE_PATH:-/opt/ic/share}"
+    NETPLAN_TEMPLATE="${NETPLAN_TEMPLATE_PATH}/99-setup-netplan.yaml.template"
+    NETPLAN_RUN_PATH="${NETPLAN_RUN_PATH:-/run/netplan}"
+    NETPLAN_OUTPUT="${NETPLAN_RUN_PATH}/99-setup-netplan.yaml"
+
+    mkdir -p "$NETPLAN_RUN_PATH"
+    cp "$NETPLAN_TEMPLATE" "$NETPLAN_OUTPUT"
+
+    sed -i "s|{INTERFACES}|${INTERFACE_LIST}|g" "$NETPLAN_OUTPUT"
+    sed -i "s|{MAC_ADDRESS}|${MAC_ADDR}|g" "$NETPLAN_OUTPUT"
+    sed -i "s|{IPV6_ADDR}|${IPV6_ADDR}|g" "$NETPLAN_OUTPUT"
+    sed -i "s|{IPV6_GATEWAY}|${IPV6_GATEWAY}|g" "$NETPLAN_OUTPUT"
+
+    # Dynamically add ethernets for each interface in descending speed
+    for IFACE in $SORTED_INTERFACES; do
+        sed -i "/^  ethernets:/a \ \ \ \ $IFACE:\n      mtu: 1500\n      optional: true\n      emit-lldp: true\n" "$NETPLAN_OUTPUT"
```

Comment:
Spawned from: https://github.com/dfinity/ic/pull/3135#issuecomment-2561675923

In our old code, we were writing the following values to systemd-networkd directly (see Systemd.rs):
```
LLDP=true
EmitLLDP=true
```

Netplan does have an LLDP property, and only an emit-lldp property. Is this sufficient?